src/coreclr/src/vm/eepolicy.cpp

```diff
@@ -498,14 +498,7 @@ void SafeExitProcess(UINT exitCode, BOOL fAbort = FALSE, ShutdownCompleteAction
         // Watson code
         CONTRACT_VIOLATION(ThrowsViolation);
 
-#ifdef TARGET_UNIX
-        if (fAbort)
-        {
-            TerminateProcess(GetCurrentProcess(), exitCode);
-        }
-#endif
-
-        EEPolicy::ExitProcessViaShim(exitCode);
+        EEPolicy::ExitProcessViaShim(exitCode, fAbort);
```

Comment:
This is the only place where `ExitProcessViaShim` is called. I would delete it and inline what you got below instead.